iphone/Classes/GeolocationModule.m

```diff
@@ -869,7 +869,10 @@ - (void)requestLocationPermissioniOS7:(id)args {
 -(void)requestLocationPermissions:(id)args
 {
     if (![TiUtils isIOS8OrGreater]) {
-        [self requestLocationPermissioniOS7:args];
+        // It is required that delegate is created and permission is presented in main thread.
+        TiThreadPerformOnMainThread(^{
+            [self requestLocationPermissioniOS7:args];
+        }, NO);
```

Comment:
Actually when we run on Studio the JS is executed in a separate thread. `<run-on-mainthread/>` is false by default. But the permission are required to be presented on main thread. 

But when we run in XCode `TI_USE_KROLL_THREAD` is not defined, which makes JS run on main thread. So it very easy to miss on background thread related issues. This fix is one of them.
This can be tested by defining and un-defining `TI_USE_KROLL_THREAD` in defines.h. Should work in both cases.